### PR TITLE
ITM-205: Get Metrics Eval server to runnable state

### PR DESCRIPTION
**Pull request by nextcen-dgemoets**
* Write place-holder code that determines when transition/probe conditions are met
  * Determines when to change scenes and when to send a probe
  * Currently just returns true if there are conditions, otherwise false
* Update action handler code:
  * Update _should_reveal_injury code and post-assessment unstructured
  * Adjust apply_treatment to change injury status to 'treated'
  Update ITMScenarioSession code:
* Add notion of current scene; reorganize to have ITMSession, ITMScenario, and ITMScene.
  * NOTE: does not merge scene-level state with scenario-level state
  * Update get_available_actions to be based on action_mappings ,end_scene_allowed, and restricted_actions.
  * Update take_action to consult condition manager for next steps (probe response and/or scene change)
* Remove scenario-specific MVP hacks
* Update with latest API/YAML/Vocab changes and re-generate model files.

TESTING NOTES:
* You should be able to run the `itm_minimal_runner` client for soartech or adept sessions without crashes.
  * Be sure to pull the latest from the `ITM-205` branch from the client repo.
* You should be able to verify changes in state based on actions, but not configured state changes based on scene changes.
* You should be able to verify the server changed scenes after each action (if there are any defined `conditions`, it changes scene).

### CODE REVIEW

**Review by kaitlyn-sharo — COMMENTED** [review]
It works! The only thing I'd consider (and it probably doesn't matter) is changing `ADM chose action END_SCENE with character None and parameters None` to just `ADM chose action END_SCENE`. Just a personal preference though. Looks great :)

File: swagger_server/itm/itm_scenario.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+from .itm_ta1_controller import ITMTa1Controller
+
+@dataclass
+class ITMScenarioData:
+    scenario: Scenario = None
+    scenes: List[ITMScene] = None
+    current_scene :ITMScene = None
+    current_scene_index = 0
+    character_simulations: List[CharacterSimulation] = None
+    character_simulator: ITMCharacterSimulator = None
```
[the remark this answers is not in the document]
Comment (nextcen-dgemoets) [reply swagger_server/itm/itm_scenario.py:17]: Yep, that's a good compromise.  I can't say that it's applied 100% consistently, though.  :)

File: swagger_server/itm/itm_scenario.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            self.scene_type
+        )
+
+    # Pass-through to ITMScene
+    def get_available_actions(self) -> List[Action]:
+        return self.isd.current_scene.get_available_actions()
+
+    def respond_to_probe(self, probe_id, choice_id, justification):
+        response = ProbeResponse(scenario_id=self.id, probe_id=probe_id, choice=choice_id,
+                                 justification = '' if justification == None else justification)
```
[the remark this answers is not in the document]
Comment (nextcen-dgemoets) [reply swagger_server/itm/itm_scenario.py]: Fixed.

**Comment by nextcen-dgemoets** [comment]
> It works! The only thing I'd consider (and it probably doesn't matter) is changing `ADM chose action END_SCENE with character None and parameters None` to just `ADM chose action END_SCENE`. Just a personal preference though. Looks great :)

Fixed.

**Review by kaitlyn-sharo — APPROVED** [review]

**Review by dereknop — APPROVED** [review]
Thought eval wasn't working but it's just because TA1 servers aren't running

**Pull request merged**